### ISSUE TITLE
Move off from VSTest so we can get test console log failures

### ISSUE THREAD

**Issue by joperezr** [dotnet/iot#2115]
Due to a bug in vstest, when a test fails and we are running them in parallel we may lose the console output making it impossible to debug the test failure, especially when we have flaky tests. This issue is to track the work of move away from vstest and use the runner directly to avoid this issue.

**Comment by ViktorHofer** [comment]
Yes, but mostly for historic reasons and because VSTest is too heavy weight for runtime's usecase. xunit.console is no longer maintained by Brad.

**Comment by ViktorHofer** [comment]
> Isn't better migrating to the same infrastructure used by the runtime?

Definitely not. Our infrastructure only exists because of corner case requirements (we build the runtime ourselves) that other repositories shouldn't need to consider. I would recommend to stick using VSTest and work with their team on a path that resolves your issues so that customers would benefit from that as well.

@Evangelink @nohwnd who would be the best person to talk to about https://github.com/microsoft/vstest/pull/2702 to prevent iot moving away from VSTest?

**Comment by Evangelink** [comment]
Thanks for the ping @ViktorHofer! @joperezr Sending you an invite for a call (will cc you @ViktorHofer).

**Comment by krwq** [comment]
[Triage] Closing this as we don't think this still repros. We can reiterate on this if needed
